membership_file/tests/test_middleware.py

```diff
@@ -0,0 +1,43 @@
+from django.contrib.auth.models import User, AnonymousUser
+from django.test import TestCase, RequestFactory
+
+from membership_file.exceptions import UserIsNotCurrentMember
+from membership_file.middleware import MembershipMiddleware
+from membership_file.models import Member
+
+
+class MembershipMiddlewareTestCase(TestCase):
+    fixtures = ['test_users', 'test_members']
+
+    def setUp(self):
+        self.middleware = MembershipMiddleware()
+
+    def _create_request(self, user_id):
+        request = RequestFactory().get('')
+        request.user = User.objects.get(id=user_id)
+        return request
+
+    def test_process_request_not_a_member(self):
+        request = self._create_request(2)
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, None)
+
+    def test_process_request_is_a_member(self):
+        request = self._create_request(100)
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, Member.objects.get(user__id=100))
+
+    def test_process_request_anonymoususer(self):
+        request = RequestFactory().get('')
+        request.user = AnonymousUser()
+        self.middleware.process_request(request)
+        self.assertEqual(request.member, None)
+
+    def test_process_exception(self):
+        request = self._create_request(1)
+        self.assertIsNone(self.middleware.process_exception(request, Exception()))  # Ignore irrelevant errors
+
+        response = self.middleware.process_exception(request, UserIsNotCurrentMember())
```

Comment:
I don’t know that much about what should and shouldn’t be tested, but is maybe useful to add a second test to check if util.py’s `MembershipRequiredMixin.dispatch()` actually generates the exceptions? (And maybe a sample page that should throw this too, but that might be too much)

(Please feel free to disagree with this one though.)

Reply:
I agree, I will do this later as I wrote testcases for the MembershipRequiredMixin in the nextcloud PR already and will update that based on this behaviour once merged

Reply:
👍